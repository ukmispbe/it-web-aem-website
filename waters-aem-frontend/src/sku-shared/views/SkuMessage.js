import React from 'react';
import ReactSVG from 'react-svg';

class SkuMessage extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <div className="cmp-notification-wrapper">
                <ReactSVG
                    src={this.props.icon}
                    className={`cmp-notification-icon`}
                />
                <div className="cmp-notification-body">
                    <div className="cmp-notification-description">
                            {this.props.message} 
                            {(this.props.linkMessage && this.props.link) &&
                                <a href={this.props.link}>{this.props.linkMessage}</a>
                            }
                    </div>
                </div>
            </div>
        )
    }
}

export default SkuMessage;
