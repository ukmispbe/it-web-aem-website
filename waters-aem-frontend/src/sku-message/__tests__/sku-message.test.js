import React from 'react';
import { shallow } from 'enzyme';
import SkuMessage from "../index";
import renderer from "react-test-renderer";

describe('Feature: SkuMessage React Component', () => {
    let props ={}, wrapper='';
    beforeEach(() => {
        props = {
            icon: "icon",
            link: "link",
            linkMessage: "linkmessage",
            message: "message"
        };
        wrapper = shallow(<SkuMessage {...props} />);
    });

    describe('Scenario: Rendering', () => {
        describe('When the component renders with props', () => {
            it('Then the snapshot should match', () => {
                const json = renderer.create(<SkuMessage {...props} />);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When all of the props are provided', () => {
            beforeEach(() => {
                props = {
                    icon: "icon",
                    link: "link",
                    linkMessage: "linkmessage",
                    message: "message"
                };
                wrapper = shallow(<SkuMessage {...props} />);
            });
            it('Then it should populate the icon', () => {
                const svgSrc = wrapper.find('ReactSVG').prop("src");
                expect(svgSrc).toEqual(props.icon);
            });

            it('Then it should populate the href with the link', () => {
                const href = wrapper.find('a').prop("href");
                expect(href).toEqual(props.link);
            });

            it('Then it should populate the A tag text with the linkMessage text', () => {
                const aTag = wrapper.find('a');
                expect(aTag.text()).toEqual(props.linkMessage);
            });

            it('Then it should populate the notification description with the message text', () => {
                const descDiv = wrapper.find('div.cmp-notification-description');
                expect(descDiv.text()).toContain(props.message);
            });

            it('Then it should populate the notification distributor description with the message text', () => {
                const descDiv = wrapper.find('div.cmp-notification-distributor');
                expect(descDiv.text()).toContain(props.distMessage);
            });
        });

        describe('When none of the props are provided', () => {
            beforeEach(() => {
                props = {
                    icon: "",
                    link: "",
                    linkMessage: null,
                    message: ""
                };
                wrapper = shallow(<SkuMessage {...props} />);
                wrapper.debug();
            });
            it('Then it should not populate the icon', () => {
                const svgSrc = wrapper.find('ReactSVG').prop("src");
                expect(svgSrc).toEqual(props.icon);
            });

            it('Then it should not render the A tag (link & linkMessage)', () => {
                const aTag = wrapper.find('a');
                expect(aTag).toEqual({});
            });

            it('Then it should not populate the notification description text', () => {
                const descDiv = wrapper.find('div.cmp-notification-description');
                expect(descDiv.text()).toBe(props.message);
            });

            it('Then it should not populate the notification description text', () => {
                const descDiv = wrapper.find('div.cmp-notification-distributor');
                expect(descDiv.text()).toBe(props.distMessage);
            });
        });
    });
});