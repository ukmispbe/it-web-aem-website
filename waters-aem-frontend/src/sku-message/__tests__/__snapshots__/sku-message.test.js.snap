// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: SkuMessage React Component Scenario: Rendering When the component renders with props Then the snapshot should match 1`] = `
<div
  className="cmp-notification-wrapper"
>
  <div
    className="cmp-notification-icon"
  />
  <div
    className="cmp-notification-body"
  >
    <div
      className="cmp-notification-description"
    >
      message
      <a
        href="link"
      >
        linkmessage
      </a>
    </div>
    <div
      className="cmp-notification-distributor"
    />
  </div>
</div>
`;
